src/electionguard/chaum_pedersen.cpp

```diff
@@ -382,22 +385,23 @@ namespace electionguard
         Log::trace("beta: ", beta->toHex());
 
         // Get our values from the precomputed values.
-        auto triple1 = precomputedTwoTriplesAndAQuad->get_triple1();
-        auto r = triple1->get_exp();
-        auto triple2 = precomputedTwoTriplesAndAQuad->get_triple2();
-        auto quad = precomputedTwoTriplesAndAQuad->get_quad();
-        auto u = triple2->get_exp();
-        auto v = quad->get_exp1();
-        auto w = quad->get_exp2();
-
-        auto a0 = triple2->get_g_to_exp();                      // 𝑔^𝑢 mod 𝑝
-        auto b0 = triple2->get_pubkey_to_exp();                 // 𝐾^𝑢 mod 𝑝
-        auto a1 = quad->get_g_to_exp1();                        // 𝑔^v mod 𝑝
-        auto b1 = quad->get_g_to_exp2_mult_by_pubkey_to_exp1(); // g^w⋅K^v mod p
+        auto triple1 = precomputedTwoTriplesAndAQuad.get_triple1();
```

Comment:
this can also probably be further optimized. it might also be worthwhile making the proof own the precomputes and the selection not own them, but this was how it shook out